src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java

```diff
@@ -139,24 +139,24 @@ private StartWorkflowExecutionParameters getStartWorkflowExecutionParameters(
     }
     p.setInput(dataConverter.toData(args));
     p.setWorkflowType(new WorkflowType().setName(workflowType.get()));
-    p.setMemo(convertMemoFromObjectToBytes(o.getMemo()));
+    p.setMemo(convertMapFromObjectToBytes(o.getMemo()));
+    p.setSearchAttributes(convertMapFromObjectToBytes(o.getSearchAttributes()));
```

Comment:
I don't think you want JSON as search attribute value.

Reply:
You are right, for now client-server requires such Json encoding, but we can remove it. https://github.com/uber/cadence/issues/2433